src/core/types.h

```diff
@@ -147,6 +148,9 @@ struct ParamNames {
     std::vector<llvm::StringRef> args;
     llvm::StringRef vararg, kwarg;
 
+    std::vector<AST_Name*> arg_names;
+    AST_Name* vararg_name, *kwarg_name;
```

Comment:
We use ```ParamNames``` also for specifying argument names for runtime functions where we don't need and don't have a ```InternedStringPool``` available which we would need to completely switch to ```AST_Name``` nodes.
Maybe it's best to split this struct into two? Or just add a comment that the AST_Names will only get set if source is available...